src/Engine/Scene/Light.hpp

```diff
@@ -103,14 +102,35 @@ class RA_ENGINE_API Light : public Component
      */
     void initialize() override;
 
+  protected:
+    Data::RenderParameters m_params;
+
   private:
     Core::Utils::Color m_color = Core::Utils::Color::White();
 
     LightType m_type { LightType::DIRECTIONAL };
 };
 
+// ---------------------------------------------------------------------------------------------
+// ---- inline methods implementation
+
+inline const Core::Utils::Color& Light::getColor() const {
+    return m_color;
+}
+
+inline void Light::setColor( const Core::Utils::Color& color ) {
+    m_color = color;
+    m_params.addParameter( "light.color", m_color );
+}
+
+inline const Light::LightType& Light::getType() const {
+    return m_type;
+}
+
+inline Data::RenderParameters& Light::getRenderParameters() const {
+    return const_cast<Data::RenderParameters&>( m_params );
```

Comment:
This is because RenderParameters only automatically binds references to other RenderParameters (and not const references, see `Ra::Engine::Data::RenderParameters::BindableTypes`. ) and some shaders (for volume rendering in ForwardRenderer but also others in external applications, extension libraries or plugins) need the parameters of the current light to be bound when rendering, without rendering a RenderObjects that manage the binding explicitly. 
So, either we return a non-const reference here, or we add automatic binding of const reference in the RenderParameters class. I'll try the second to check the impact of this choice and will push my proposal ASAP.